modules/rostests/apitests/shell32/StrRStr.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Test for StrRStrA/W
+ * COPYRIGHT:   Copyright 2025 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include "shelltest.h"
+#include <versionhelpers.h>
+
+typedef PSTR (WINAPI *FN_StrRStrA)(PCSTR, PCSTR, PCSTR pszSearch);
+typedef PWSTR (WINAPI *FN_StrRStrW)(PCWSTR, PCWSTR, PCWSTR pszSearch);
+
+static VOID TEST_StrRStrA(VOID)
+{
+    PCSTR psz, pch;
+    PSTR ret;
+    FN_StrRStrA StrRStrA = (FN_StrRStrA)GetProcAddress(GetModuleHandleW(L"shell32"), MAKEINTRESOURCEA(389));
+
+    if (!StrRStrA)
+    {
+        skip("StrRStrA not found\n");
+        return;
+    }
+
+    ret = StrRStrA("ABCBC", NULL, "BC");
+    ok_str(ret, "BC");
```

Comment:
If I were you I wouldn't do the checks like that, but instead, check whether the result points at the correct place into the source string.
Like this:
```suggestion
    psz = "ABCBC";
    ret = StrRStrA(psz, NULL, "BC");
    ok_ptr(ret, psz + 3);
```
etc.

Reply:
Improved.